src/cpc/cpc.c

```diff
@@ -115,26 +115,43 @@ static unsigned long   NopCount = 0;
 /* DOS rom is NULL if no DOS is on the motherboard */
 /* basic rom */
 static const unsigned char *pBasic;
+/* An allocated buffer for the ROM, so they can be overridden */
+static unsigned char *pBasic_ROM;
 
 /* os rom */
 const unsigned char *pOS;
+/* An allocated buffer for the ROM, so they can be overridden */
+static unsigned char *pOS_Rom;
 
 /* dos rom */
 static const unsigned char *pDOS;
+/* An allocated buffer for the ROM, so they can be overridden */
+static unsigned char *pDOS_Rom;
 
 void CPC_SetOSRom(const unsigned char *pOSRom)
 {
-	pOS = pOSRom;
+	free(pOS_Rom);
+	pOS_Rom = malloc(16384);
+	memcpy(pOS_Rom, pOSRom, 16384);
```

Comment:
the choice of names here is pretty confusing. at first i thought you're memcpying the same variable onto itself.